sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1150,12 +1166,11 @@ case class StringTrim(srcStr: Expression, trimStr: Option[Expression] = None)
 
   override protected def direction: String = "BOTH"
 
-  override def doEval(srcString: UTF8String): UTF8String = srcString.trim()
+  override def doEval(srcString: UTF8String): UTF8String =
+    CollationSupport.StringTrim.exec(srcString, collationId)
 
   override def doEval(srcString: UTF8String, trimString: UTF8String): UTF8String =
-    srcString.trim(trimString)
-
-  override val trimMethod: String = "trim"
```

Comment:
why are you removing this?

Reply:
It was only used in `doGenCode` in the superclass.

After we changed to use `CollationSupport` classes/methods, and not call methods from `UTF8String` directly, there is no need for this field anymore.

Earlier this field was used in `doGenCode` to call specific method of `UTF8String`. Now, if we kept same behavior to call methods from `CollationSupport` we would either need to have a lot of hardcoded string compares or use reflection to invoke methods.

I think that this behavior where we match classes in `doGenCode` and then call respective methods from `CollationSupport` is more performant and robust.

Reply:
got it, sounds good